core/src/main/java/com/linecorp/armeria/server/healthcheck/SettableHealthStatusChecker.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2015 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.healthcheck;
+
+import javax.annotation.Nonnull;
+
+import com.linecorp.armeria.common.util.AbstractListenable;
+import com.linecorp.armeria.server.Server;
+
+/**
+ * A simple {@link ListenableHealthChecker} whose state can be set by a caller. This can be used in case server
+ * health should have additional conditions besides the state of the {@link Server}. e.g. it should depend on
+ * the health of a backend.
+ */
+public final class SettableHealthStatusChecker extends AbstractListenable<HealthChecker>
```

Comment:
- Can we merge this into `SettableHealthChecker`? Is there any reason we have to keep both `SettableHealthChecker` and `SettableHealthStatusChecker`?
- There are no tests for this class.

Reply:
>Can we merge this into SettableHealthChecker? Is there any reason we have to keep both SettableHealthChecker and SettableHealthStatusChecker?

Yes, we can merge it to `SettableHealthChecker`. I separated the classes because I thought the code would be too messy to maintain backward compatibility of `SettableHealthChecker` and add new feature(setting `HealthStatus`). However, I think it's possible to maintain both backward compatibility and adding new feature without making `SettableHealthChecker` too messy. 

>There are no tests for this class.

I'm gonna remove `SettableHealthStatusChecker`, so I'll add some test into `SettableHealthChecker`. Thank you for the through review.

Reply:
Thanks!